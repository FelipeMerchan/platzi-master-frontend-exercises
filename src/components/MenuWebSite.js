import React, { useState, useEffect } from 'react'
import { getRandomColor } from '../utils/getRandomColor'

import './styles/MenuWebSite.css'

function MenuWebSite() {
  const [count, setCount] = useState(0)
  const composeEvent = new Event('composed')


  const handleClick = (e, composeEvent, getRandomColor) => {
    e.preventDefault()
    const { target: { id } } = e
    const anchorActive = document.querySelector('.is-active')

    anchorActive.classList.remove('is-active')
    e.target.classList.add('is-active')
    document.documentElement.style.setProperty('--primaryColor', getRandomColor())

    e.target.dispatchEvent(composeEvent);

    setCount(count + 1)
    console.log(`Id: ${id}, Count:${count}`)
  }

  useEffect(() => {
    const links = document.querySelectorAll('.MenuWebSite_item > a')

    links.forEach(object => {
      object.addEventListener('composed', function(e) {
        e.stopPropagation()
        console.log('IÂ´m not propagating')
      })
    })
  })

  return (
    <nav className="MenuWebSite">
      <ul>
        <li className="MenuWebSite_item">
          <a className="is-active" role="button" onClick={ (e) => handleClick(e, composeEvent, getRandomColor) } href="#" id="home">Home</a>
        </li>
        <li className="MenuWebSite_item">
          <a role="button" onClick={ (e) => handleClick(e, composeEvent, getRandomColor) } href="#" id="aboutUs">Acerca de</a>
        </li>
        <li className="MenuWebSite_item">
          <a role="button" onClick={ (e) => handleClick(e, composeEvent, getRandomColor) } href="#" id="portfolio">Portafolio</a>
        </li>
        <li className="MenuWebSite_item">
          <a role="button" onClick={ (e) => handleClick(e, composeEvent, getRandomColor) } href="#" id="blog">Blog</a>
        </li>
        <li className="MenuWebSite_item">
          <a role="button" onClick={ (e) => handleClick(e, composeEvent, getRandomColor) } href="#" id="contactUs">Contacto</a>
        </li>
      </ul>
    </nav>
  )
}

export default MenuWebSite