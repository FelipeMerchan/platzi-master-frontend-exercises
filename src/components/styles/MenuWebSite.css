.MenuWebSite {
  height: 50px;
  margin: var(--space) 0 var(--space-x2);
  overflow: hidden;
  background-color: var(--backgroundDarkGray);
  border-radius: var(--border-radius);
}

.MenuWebSite ul {
  display: flex;
  align-items: center;
  width: 100%;
  height: inherit;
  padding: 0;
  margin: 0;
}

.MenuWebSite ul li {
  flex: 1;
  height: inherit;
  line-height: 50px;
  text-align: center;
  list-style-type: none;
}

.MenuWebSite ul li a {
  display: flex;
  justify-content: center;
  font-weight: 700;
  text-decoration: none;
  color: var(--redColor);
}

.MenuWebSite ul li a:hover,
.MenuWebSite ul li a:active {
  color: #ffffff;
  background-color: var(--redColor);
}

@media screen and (max-width: 480px) {
  .MenuWebSite {
    height: auto;
  }

  .MenuWebSite ul{
    display: initial;
  }
}