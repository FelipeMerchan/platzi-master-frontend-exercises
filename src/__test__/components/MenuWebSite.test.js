import React from 'react'
import { shallow, mount } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import MenuWebSite from '../../components/MenuWebSite'

describe('Probar el componente <MenuWebSite />', () => {
  const menuWebSite = shallow(<MenuWebSite />)
  test('Render del componente MenuWebSite', () => {
    expect(menuWebSite.length).toEqual(1)
  })
  test('Render con el className MenuWebSite', () => {
    expect(menuWebSite.find('nav').hasClass('MenuWebSite')).toBe(true)
  })
})

describe('Probar el estado active del componente MenuWebSite', () => {
  const handleClick = jest.fn();
  const menuWebSite = mount(<MenuWebSite />)
  const thirdButton = menuWebSite.find('a').at(3)
  test('Comprobar los clics en los botones de navegaciÃ³n', () => {
    thirdButton.simulate('click');
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
  test('Render con el className is-active', () => {
    expect(menuWebSite.exists('.is-active')).toBeTruthy()
  })
})

describe('MenuWebSite Snapshot', () => {
  test('Comprobar la UI del componente MenuWebSite', () => {
    const menuWebSite = create(<MenuWebSite />)
    expect(menuWebSite.toJSON()).toMatchSnapshot()
  })
})