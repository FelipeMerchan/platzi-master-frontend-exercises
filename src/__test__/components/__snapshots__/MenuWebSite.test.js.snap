// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuWebSite Snapshot Comprobar la UI del componente MenuWebSite 1`] = `
<nav
  className="MenuWebSite"
>
  <ul>
    <li
      className="MenuWebSite_item"
    >
      <a
        className="is-active"
        href="#"
        id="home"
        onClick={[Function]}
        role="button"
      >
        Home
      </a>
    </li>
    <li
      className="MenuWebSite_item"
    >
      <a
        href="#"
        id="aboutUs"
        onClick={[Function]}
        role="button"
      >
        Acerca de
      </a>
    </li>
    <li
      className="MenuWebSite_item"
    >
      <a
        href="#"
        id="portfolio"
        onClick={[Function]}
        role="button"
      >
        Portafolio
      </a>
    </li>
    <li
      className="MenuWebSite_item"
    >
      <a
        href="#"
        id="blog"
        onClick={[Function]}
        role="button"
      >
        Blog
      </a>
    </li>
    <li
      className="MenuWebSite_item"
    >
      <a
        href="#"
        id="contactUs"
        onClick={[Function]}
        role="button"
      >
        Contacto
      </a>
    </li>
  </ul>
</nav>
`;
